rcap/requesthandler.go

```diff
@@ -27,6 +27,7 @@ type RequestHandlerConfig struct {
 // RequestHandlerCapability allows runnables to handle HTTP requests
 type RequestHandlerCapability interface {
 	GetField(fieldType int32, key string) ([]byte, error)
+	SetField(fieldType int32, key string, val string) error
 	SetResponseHeader(key, val string) error
 }
 
```

Comment:
Like the existing `CacheCapability`, it would be good to have granular rules for each of the functions.

```go
type RequestHandlerRules struct {
	AllowSetField          bool
	AllowGetField          bool
	AllowSetResponseHeader bool
}
```

Alternatively, maybe something like this:
```go
type RequestHandlerMask uint64

const (
	GetField RequestHandlerMask = 1 << iota
	SetField
	SetResponseHeader
)

const (
	GetPermissions RequestHandlerMask = GetField
	SetPermissions                    = SetField | SetResponseHeader
	AllPermissions                    = GetPermissions | SetPermissions
)
```
This way it would be easy to pass and use a single permissions mask back and forth across the FFI boundary if we ever needed to. For example, the helper libs could provide better error messages from within the module itself.

Getting ahead of myself now, but this would be neat to do code generation for as well.